### PR TITLE
Added error handling for observation API endpoints

### LINKED ISSUE #160: Refactor observation endpoints
- Add error handling
- Update /api/observation/create to return the new observation
- Pass all tests

### PR DESCRIPTION
**Pull request by kevhuang**
Also modified the /api/observation/create endpoint to return the new observation instead of an array of observations

Resolves #160

### CODE REVIEW

File: server/api/observation/index.js [hunk 1]
```diff
@@ -3,60 +3,72 @@ var db = require('../../db');
...
-   content: req.body.content || '',
-   completed: req.body.completed || false,
-   rawImage: req.body.rawImage || undefined,
-   UserId: req.body.userId,
-   EventId: req.body.eventId
- }).then(function(observation){
-    //Find the event to which the observation belongs to
-    db.Event.findOne({
+  // Check if userId and eventId exist and are numbers
+  if (req.body.userId && !isNaN(req.body.userId) && req.body.eventId && !isNaN(req.body.eventId)) {
```
Comment (ekong2) [inline server/api/observation/index.js]: I think checking for !isNaN(req.body.userId) && !isNaN(req.body.userId) might be enough

**Comment by ekong2** [comment]
:+1: solid work

File: server/api/observation/index.js (hunk 1 above)
Comment (kevhuang) [inline server/api/observation/index.js]: Excellent point...I will refactor..thanks!

**Comment by dmsakamoto** [comment]
:+1:

**Pull request merged**